code/datums/ftl_ships/ship_weapons.dm

```diff
@@ -146,6 +146,33 @@
 	playsound(epicenter, 'sound/magic/lightningbolt.ogg', 100, 1)
 	empulse(epicenter,5,10,1)
 
+
+/datum/ship_attack/carrier_weapon
+	cname = "Carrier Blaster"
+	projectile_effect = "leaper"
+	hull_damage = 0
+	shield_bust = 1
+	var/list/boarding_mobs = list(/mob/living/simple_animal/hostile/droid)
+	var/amount = 5
+
+/datum/ship_attack/carrier_weapon/damage_effects(turf/epicenter)
+
+	playsound(epicenter, 'sound/ftl/shipweapons/carrier_hit.ogg', 100, 1)
+	for(var/I = 1 to amount)
+		var/path = pick(boarding_mobs)
+		var/mob/to_spawn = new path(epicenter)
+		to_spawn.faction = list(ship.faction)
+
+/datum/ship_attack/carrier_weapon/oneTime
+	var/fired = FALSE
+
+/datum/ship_attack/carrier_weapon/oneTime/damage_effects(turf/epicenter)
+	if(!fired)
+		..()
+		fired = TRUE
```

Comment:
looking back at this code, i'm probably going to add the onetime variable and another one to the parent instead, so you can have the code both look for it firing once and then have a specific variable to have it go through the list and spawn all of the mobs specifically instead of still doing it random like it is here. I'd like to do this in another PR though, since this one is essentially fully functional.